orchagent/orch.cpp

```diff
@@ -310,66 +310,50 @@ bool Orch::bake()
 - both type_name and object_name are cleared to empty strings as an
 - indication to the caller of the special case
 */
-bool Orch::parseReference(type_map &type_maps, string &ref_in, string &type_name, string &object_name)
+bool Orch::parseReference(type_map &type_maps, string &ref_in, const string &type_name, string &object_name)
 {
     SWSS_LOG_ENTER();
 
     SWSS_LOG_DEBUG("input:%s", ref_in.c_str());
-    if (ref_in.size() < 2)
-    {
-        SWSS_LOG_ERROR("invalid reference received:%s\n", ref_in.c_str());
-        return false;
-    }
-    if ((ref_in[0] != ref_start) || (ref_in[ref_in.size()-1] != ref_end))
-    {
-        SWSS_LOG_ERROR("malformed reference:%s. Must be surrounded by [ ]\n", ref_in.c_str());
-        return false;
-    }
-    if (ref_in.size() == 2)
+
+    if (ref_in.size() == 0)
     {
-        // value set by user is "[]"
+        // value set by user is ""
         // Deem it as a valid format
         // clear both type_name and object_name
         // as an indication to the caller that
         // such a case has been encountered
-        type_name.clear();
```

Comment:
Can an empty string be passed by redis-db? If no and this is a valid flow, we need to find another way to indicate clearing reference. Probably we can use "NULL".